beacon_node/beacon_chain/tests/store_tests.rs

```diff
@@ -1012,6 +1012,106 @@ fn pruning_does_not_touch_blocks_prior_to_finalization() {
     assert!(harness.chain.knows_head(&stray_head));
 }
 
+#[test]
+fn pruning_does_not_touch_blocks_prior_to_finalization_yoke() {
+    const HONEST_VALIDATOR_COUNT: usize = 16;
+    const ADVERSARIAL_VALIDATOR_COUNT: usize = 8;
+    let mut yoke = BeaconChainYoke::new(
+        MinimalEthSpec,
+        HONEST_VALIDATOR_COUNT,
+        ADVERSARIAL_VALIDATOR_COUNT,
+    );
+    let slots_per_epoch = yoke.slots_per_epoch();
+
+    let mut state = yoke.get_current_state();
+
+    // Fill up 0th epoch with canonical chain blocks
+    let zeroth_epoch_slots: Vec<Slot> = (1..=slots_per_epoch).map(Slot::new).collect();
+    let (canonical_chain_blocks, _, _, new_state) = yoke.add_attested_blocks_at_slots(
+        state,
+        &zeroth_epoch_slots,
+        &yoke.get_honest_validators(),
+    );
+    state = new_state;
+    let canonical_chain_slot: u64 = yoke.get_current_slot().into();
+
+    // Fill up 1st epoch.  Contains a fork.
+    let first_epoch_slots: Vec<Slot> = ((slots_per_epoch + 1)..(2 * slots_per_epoch))
+        .map(Slot::new)
+        .collect();
+    let (stray_blocks, stray_states, stray_head, _) = yoke.add_attested_blocks_at_slots(
+        state.clone(),
+        &first_epoch_slots,
+        &yoke.get_adversarial_validators(),
+    );
+
+    // Preconditions
+    for &block_hash in stray_blocks.values() {
+        let block = yoke.get_block(block_hash);
+        assert!(
+            block.is_some(),
+            "stray block {} should be still present",
+            block_hash
+        );
+    }
+
+    for (&slot, &state_hash) in &stray_states {
+        let state = yoke.get_state(state_hash);
+        assert!(
+            state.is_some(),
+            "stray state {} at slot {} should be still present",
+            state_hash,
+            slot
+        );
+    }
+
+    assert_eq!(
+        yoke.get_finalized_checkpoints_hashes(),
+        [Hash256::zero().into()].iter().cloned().collect(),
+    );
+
+    // Trigger finalization
```

Comment:
This seems like it doesn't trigger finalization because there are too few validators (need >2/3, have 2/3), and the `slots_per_epoch` block that gets finalized only gets finalized because the fork starts after it. Based on the name of the test, this isn't what I was expecting (I thought finalization would be triggered and the fork would be confirmed deleted). I think the name/comments could do with clarification, or some parameters tweaked so that it does what I thought it would

Reply:
> the slots_per_epoch block that gets finalized only gets finalized because the fork starts after it

Could you elaborate how forks influence finalization?

The [paper](https://arxiv.org/abs/2003.03052) talks about supermajority as 2/3, not 2/3+1 or more.

Reply:
You're right, I misremembered. I thought it was the same as old-school `3f + 1` BFT algorithms like PBFT, but it ain't!